### PR TITLE
Add new rule NoPartialFunctions

### PR DESCRIPTION
**Pull request by jgardella**
Adds a new rule to disallow partial functions. The rule will suggest to use non-partial functions or pattern matching instead.

### CODE REVIEW

File: src/FSharpLint.Core/Rules/Conventions/NoPartialFunctions.fs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+type Config = {
+    AllowedPartials:string list
+    AdditionalPartials:string list
+}
+
+type private Replacement =
+    | PatternMatch
+    | Function of functionName:string
+
+let private partialFunctionIdentifiers =
```
Comment (jgardella) [inline src/FSharpLint.Core/Rules/Conventions/NoPartialFunctions.fs:20]: @Krzysztof-Cieslak I added all the partial collection functions that I'm aware of, anything else we should add?

Comment (knocte) [reply src/FSharpLint.Core/Rules/Conventions/NoPartialFunctions.fs:20]: @jgardella how about Map.Item -> Map.TryFind

**Comment by knocte** [comment]
who needs to review this? would be nice to get it merged

**Comment by duckmatt** [comment]
Just tried to resolve the conflicts in the Github UI, was not a good idea 😅

**Comment by duckmatt** [comment]
@knocte Looks good to go in.

Potentially worth considering having these be hints although there is the downside which is that we'd lose the applicable documentation page

File: src/FSharpLint.Core/Rules/Conventions/NoPartialFunctions.fs (hunk 1 above)
Comment (duckmatt) [reply src/FSharpLint.Core/Rules/Conventions/NoPartialFunctions.fs:20]: @knocte At the moment it'll require a little bit more work to handle methods on an object instance (some type checking) - might be worth another PR for that, we could potentially add:

`Map.find` -> `Map.tryFind`
`Map.findKey` -> `Map.tryFindKey`

**Pull request merged**